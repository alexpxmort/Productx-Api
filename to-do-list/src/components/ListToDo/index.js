import { Grid } from '@material-ui/core';
import React from 'react';
import './index.scss';
import CheckCircleIcon from '@material-ui/icons/CheckCircle';
import EditIcon from '@material-ui/icons/Edit';
import RemoveIcon from '@material-ui/icons/Delete';



export default function ListToDo({items,edit,complete,deleteProp}) {

    return (
       <Grid
       id='grid_container_items'
       style={{paddingTop:'1rem'}}
       >
            {
                items.map((val,idx)=>(
                    <div key={idx} className={`item ${val.completed ? "completed":""}`}>
                       <span>{val.name}</span>
                       <CheckCircleIcon className='check' style={{color:'#ff6c6c'}} onClick={() => complete(val)}/>
                       <EditIcon className='edit' onClick={() => edit(val)}/>
                       <RemoveIcon className='remove' onClick={() => deleteProp(val)}/>
                    </div>
                ))
            }
       </Grid>
    )
}
