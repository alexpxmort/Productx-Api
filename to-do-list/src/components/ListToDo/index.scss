
.item{
    width: 100%;
    max-height: 30px;
    padding:0.4rem;
    margin: .60rem;
    display: flex;
    justify-content: flex-start;
    border: 1px solid #ccc;
    border-radius: .5rem;
    flex-direction: row;
    align-items: flex-start;
    flex-wrap: wrap;

    span{
        margin-right: 0.4rem;
    }

    .MuiSvgIcon-root:hover{
        cursor: pointer;
    }

}

.completed{
    text-decoration-style: solid;
    text-decoration-line: line-through;
    text-decoration-color: darkblue;
    opacity: 0.6;
}