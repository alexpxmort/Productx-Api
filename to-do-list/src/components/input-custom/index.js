/**
 * Componente Customizado de Input
 *
 */

import React, {useRef, useEffect} from 'react';
import {TextField, FormGroup} from '@material-ui/core';
import {useField} from '@unform/core';
import {Alert} from '@material-ui/lab';


export const InputCustom = ({
  name, label, variant, ...otherProps
}) => {
  const inputRef = useRef(null);
  const {
    fieldName, registerField, error, defaultValue,
  } = useField(name);

  useEffect(() => {
    registerField({
      name: fieldName,
      ref: inputRef.current,
      path: 'value',
    });
  }, [fieldName, registerField]);

  return (
		<div>
			  <FormGroup style={{marginTop: 10, marginBottom: 10, width: 500}}>
            <TextField label={label} variant={variant} defaultValue={defaultValue} {...otherProps} inputRef={inputRef} />
     </FormGroup>
				{error && <Alert severity="warning">{error}</Alert>}
		</div>
  );
};
