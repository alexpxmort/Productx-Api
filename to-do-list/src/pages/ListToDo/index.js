import { Grid } from '@material-ui/core';
import React from 'react';
import './index.scss';
import CheckCircleIcon from '@material-ui/icons/CheckCircle';
import EditIcon from '@material-ui/icons/Edit';
import RemoveIcon from '@material-ui/icons/Delete';



export default function ListToDoPage({items,edit,complete,deleteProp}) {

    return (
       <Grid
       style={{paddingTop:'1rem'}}
       >
            {
                items.map((val,idx)=>(
                    <div key={idx} className={`item ${val.completed ? "completed":""}`}>
                       <span>{val.name}</span>
                       <CheckCircleIcon style={{color:'#ff6c6c'}} onClick={() => complete(val)}/>
                       <EditIcon onClick={() => edit(val)}/>
                       <RemoveIcon onClick={() => deleteProp(val)}/>
                    </div>
                ))
            }
       </Grid>
    )
}
