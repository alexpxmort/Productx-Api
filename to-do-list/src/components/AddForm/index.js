import React,{forwardRef} from 'react';
import { Button } from '@material-ui/core';
import {Form} from '@unform/web';
import AddIcon from '@material-ui/icons/Add';
import { InputCustom } from '../input-custom';

const AddForm = ({handleSubmit,value},ref) => {

  return (
    <Form id='addForm' ref={ref}  initialData={{name:value || ''}} onSubmit={handleSubmit}>
      <InputCustom  className='custom-input' name={'name'} placeholder={'Name'}   />
      <Button  type="submit" variant="contained" color="primary">
        <AddIcon/>  Salvar
      </Button>
    </Form>   
  )
};
export default forwardRef(AddForm);
