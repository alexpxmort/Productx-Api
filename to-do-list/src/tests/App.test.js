import React from 'react';
import {shallow} from 'enzyme';
import {uuid as v4} from 'uuidv4';
import AddForm from '../components/AddForm';
import { InputCustom } from '../components/input-custom';
import { Form } from '@unform/web';
import ListToDo from '../components/ListToDo';
import AddToDoPage from '../pages/AddToDo';


describe ('#Components', () => {
  describe ('#AddForm', () => {
    it('Submit the form correctly',()=>{
      const ref = React.createRef();
      const _handleSubmit = jest.fn();
      let wrapper = shallow(<AddForm  handleSubmit={_handleSubmit} ref={ref}/>);
      let form = wrapper.find("#addForm");
      expect(form).toHaveLength(1);

      let nameInput = form.find('.custom-input');
      expect(nameInput).toHaveLength(1);
      expect(nameInput.getElement()['props']['name']).toEqual('name');
      expect(nameInput.getElement()['props']['placeholder']).toEqual('Name');
    
      form.simulate('submit',{preventDefault(){}});

      expect(_handleSubmit).toHaveBeenCalled();
      expect(wrapper).toMatchSnapshot();
    })
  }); 

  describe ('#InputCustom', () => {
    it('Changes the input correctly', () => {
      let wrapper = shallow(<Form initialData={{teste:''}} id='testForm'>
        <InputCustom className='teste' name={'teste'}  placeholder={'Name'}/>
      </Form>);

      expect(wrapper).toHaveLength(1);

      expect(wrapper.props()['initialData']).toEqual({teste:''})
    });
  });

  describe ('#ListToDo', () => {
    it('Renders the ListToDo correctly', async () => {
      const _delProp = jest.fn();
      const _complete = jest.fn();
      const _edit = jest.fn();
      const items = [
        {
          id:v4(),
          name:'Teste',
          created_at:new Date(),
          updated_at:new Date(),
          completed:false
        },
        {
          id:v4(),
          name:'Teste 2',
          created_at:new Date(),
          updated_at:new Date(),
          completed:false
        }
      ]
      const wrapper = shallow(<ListToDo items={items} edit={_edit} complete={_complete} deleteProp={_delProp}/>);
      expect(wrapper).toHaveLength(1);

      const container = wrapper.find('#grid_container_items');

      expect(container).toHaveLength(1);

      expect(container.children()).toHaveLength(2);

      expect(container.find('.item').at(0).children()).toHaveLength(4);
      expect(container.find('.item').at(0).find('span').text()).toEqual(items[0].name);
      expect(container.find('.item').at(0).children().get(1).type['render']['muiName']).toEqual('SvgIcon');
      expect(container.find('.item').at(0).children().get(2).type['render']['muiName']).toEqual('SvgIcon');
      expect(container.find('.item').at(0).children().get(3).type['render']['muiName']).toEqual('SvgIcon');

      expect(container.find('.item').at(1).children()).toHaveLength(4);
      expect(container.find('.item').at(1).find('span').text()).toEqual(items[1].name);
      expect(container.find('.item').at(1).children().get(1).type['render']['muiName']).toEqual('SvgIcon');
      expect(container.find('.item').at(1).children().get(2).type['render']['muiName']).toEqual('SvgIcon');
      expect(container.find('.item').at(1).children().get(3).type['render']['muiName']).toEqual('SvgIcon');

      //calls complete function item 1

      container.find('.item').at(0).find('.check').at(0).simulate('click');
      expect(_complete).toHaveBeenCalled();

      //calls edit function item 1

      container.find('.item').at(0).find('.edit').at(0).simulate('click');
      expect(_edit).toHaveBeenCalled();

      //calls remove function item 1

      container.find('.item').at(0).find('.remove').at(0).simulate('click');
      expect(_delProp).toHaveBeenCalled();

    });
  });
});

describe ('#Pages', () => {
  describe ('#AddToDoPage', () => {
    it('Renders the ListToDo correctly',() => {
      const items = [
        {
          id:v4(),
          name:'Teste',
          created_at:new Date(),
          updated_at:new Date(),
          completed:false
        },
        {
          id:v4(),
          name:'Teste 2',
          created_at:new Date(),
          updated_at:new Date(),
          completed:false
        }
      ]

      localStorage.setItem('tasks',JSON.stringify(items));
      let wrapper = shallow(<AddToDoPage/>);
      expect(wrapper).toHaveLength(1);
  
      let formComponent = wrapper.find('#formComponent');
      expect(formComponent).toHaveLength(1);
  
      let listComponent = wrapper.find('#listComponent');
      expect(listComponent).toHaveLength(1);

    })
  });
});