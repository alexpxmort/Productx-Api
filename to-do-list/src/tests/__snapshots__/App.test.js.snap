// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Components #AddForm Submit the form correctly 1`] = `
<ForwardRef
  id="addForm"
  initialData={
    Object {
      "name": "",
    }
  }
  onSubmit={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "preventDefault": [Function],
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <InputCustom
    className="custom-input"
    name="name"
    placeholder="Name"
  />
  <WithStyles(ForwardRef(Button))
    color="primary"
    type="submit"
    variant="contained"
  >
    <Memo(ForwardRef(AddIcon)) />
      Salvar
  </WithStyles(ForwardRef(Button))>
</ForwardRef>
`;
